bundles/org.dataflowanalysis.analysis.converter/src/org/dataflowanalysis/analysis/converter/PalladioConverter.java

```diff
@@ -0,0 +1,188 @@
+package org.dataflowanalysis.analysis.converter;
+
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.dataflowanalysis.analysis.DataFlowConfidentialityAnalysis;
+import org.dataflowanalysis.analysis.core.*;
+import org.dataflowanalysis.analysis.pcm.PCMDataFlowConfidentialityAnalysisBuilder;
+import org.dataflowanalysis.analysis.pcm.core.AbstractPCMActionSequenceElement;
+import org.dataflowanalysis.analysis.pcm.core.seff.SEFFActionSequenceElement;
+import org.dataflowanalysis.analysis.pcm.core.user.*;
+import org.dataflowanalysis.analysis.testmodels.Activator;
+import org.palladiosimulator.pcm.core.entity.Entity;
+
+import org.dataflowanalysis.dfd.datadictionary.*;
+import org.dataflowanalysis.dfd.dataflowdiagram.*;
+
+public class PalladioConverter extends Converter {
+
+    private final Map<Entity, Node> dfdNodeMap = new HashMap<>();
+    private final DataDictionary dataDictionary = datadictionaryFactory.eINSTANCE.createDataDictionary();
+    private final DataFlowDiagram dataFlowDiagram = dataflowdiagramFactory.eINSTANCE.createDataFlowDiagram();
+
+    private DataFlowDiagramAndDictionary processPalladio(List<ActionSequence> actionSequences) {
+        for (ActionSequence actionSequence : actionSequences) {
+            Node previousNode = null;
+            for (AbstractActionSequenceElement<?> actionSequenceElement : actionSequence.getElements()) {
+                if (actionSequenceElement instanceof AbstractPCMActionSequenceElement) {
+                    previousNode = processActionSequenceElement((AbstractPCMActionSequenceElement<?>) actionSequenceElement, previousNode);
+                }
+            }
+        }
+        return new DataFlowDiagramAndDictionary(dataFlowDiagram, dataDictionary);
+    }
+
+    private Node processActionSequenceElement(AbstractPCMActionSequenceElement<? extends Entity> pcmASE, Node previousDFDNode) {
+        Node dfdNode = getDFDNode(pcmASE);
+
+        createFlowBetweenPreviousAndCurrentNode(previousDFDNode, dfdNode, pcmASE);
+
+        return dfdNode;
+    }
+
+    private void createFlowBetweenPreviousAndCurrentNode(Node source, Node dest, AbstractPCMActionSequenceElement<? extends Entity> pcmASE) {
+        if (source == null || dest == null) {
+            return;
+        }
+        List<DataFlowVariable> flowVariables = pcmASE.getAllDataFlowVariables();
+        for (DataFlowVariable flowVariable : flowVariables) {
+            String flowName = flowVariable.variableName();
+
+            dataFlowDiagram.getFlows().stream().filter(f -> f.getSourceNode().equals(source)).filter(f -> f.getDestinationNode().equals(dest))
+                    .filter(f -> f.getEntityName().equals(flowName)).findFirst().orElse(createFlow(source, dest, flowName));
```

Comment:
Code style: Could you put each stream expression on a separate line? This would make this stream expression a bit more readable

Reply:
This is my eclipse auto formatter